samples/messagemutators/Core_8/Sample/TransportMessageCompressionMutator.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.IO;
+using System.IO.Compression;
+using System.Threading.Tasks;
+using NServiceBus.Logging;
+using NServiceBus.MessageMutator;
+
+#region TransportMessageCompressionMutator
+
+public class TransportMessageCompressionMutator :
+    IMutateIncomingTransportMessages,
+    IMutateOutgoingTransportMessages
+{
+    static ILog log = LogManager.GetLogger("TransportMessageCompressionMutator");
+
+    public Task MutateOutgoing(MutateOutgoingTransportMessageContext context)
+    {
+        log.Info($"transportMessage.Body size before compression: {context.OutgoingBody.Length}");
+
+        var mStream = new MemoryStream(context.OutgoingBody.ToArray());
+        var outStream = new MemoryStream();
+
+        using (var tinyStream = new GZipStream(outStream, CompressionMode.Compress))
+        {
+            mStream.CopyTo(tinyStream);
+        }
+        // copy the compressed buffer only after the GZipStream is disposed,
+        // otherwise, not all the compressed message will be copied.
+        context.OutgoingBody = outStream.ToArray();
+        context.OutgoingHeaders["IWasCompressed"] = "true";
+        log.Info($"transportMessage.Body size after compression: {context.OutgoingBody.Length}");
+        return Task.CompletedTask;
+    }
+
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("NServiceBus.Code", "NSB0002:Forward the 'CancellationToken' property of the context parameter to methods", Justification = "<Pending>")]
```

Comment:
This is here to prevent needing an `#ifdef` for line 45. .NET 4.8 does not have the cancellation token overload while the rest of the runtimes do